meta/engineering_principles.md

```diff
@@ -0,0 +1,159 @@
+changes.
+
+For meetings over 2-3 people, provide agendas and make as many people optional as possible. We've seen great
+success on automation via documentation for recurring meetings.
+
+- Examples
+
+  - Danger
+  - Peril
+  - RFC Process
+  - artsy/meta
+
+- Further Reading
+
+  - http://artsy.github.io/blog/2018/05/25/support-process/
+  - http://artsy.github.io/blog/2017/09/04/Introducing-Peril/
+  - http://artsy.github.io/blog/2018/02/02/artsy-apogee/
+
+### Leverage Your Impact
+
+We built an outsized reputation for our size as an engineering team via the other principles. This gives the
+ability to move local impact to a larger communal impact.
+
+Consider ways in which you can expand the scope of impact for your time:
+
+- Avoid private messaging when you could communicate in a public channel
+- Team presentations could be worked for public consumption
+- Internal notes could be turned into blog posts
+- Documentation on teams and processes could be open for others to reference and learn from
+- A regular meetup could be converted into a franchisable system for others to replicate
+
+The small improvements eventually add up so that you can do industry level work as a part of building Artsy.
+
+- Examples
+
+  - Artsy x React Native
+  - JavaScriptures
+  - Learn Swift at Artsy
+  - Artsy Blog
+  - artsy/meta
+  - Peer Labs
+  - http://artsy.github.io/blog/2015/04/03/artsy-engineering-compensation-framework/
+
+### De-silo Engineers
+
+Provide people the ability to work in many spaces and not feel blocked by another team. Common languages, common
+terminology and common idioms. There's definitely space for experimentation, but within a shared framework. The aim
+is not to stifle creativity, but increase individual scope.
+
+- Examples
+
+  - Moving to React Native
+  - Teach Swift via Swift at Artsy
+  - Migrating Admin tools to shared Front-End stacks
+  - Consolidated Analytics schema
+  - Dev Swap which switches teams and responsibilities
+
+- Further Reading
+  - https://www.youtube.com/watch?v=hlLhtWLghGA
+  - http://artsy.github.io/blog/2017/09/11/DevSwap/
+  - http://artsy.github.io/artsy-x-react-native.html
+
+### Build for 10x
+
+A system doesn't live forever, but you should make decisions with a aim to handle 10x the traffic, number of
+contributors and company revenue. Basically a single order of magnitude of growth everywhere. This can help guide
+deciding on technological choices and figuring the trade-offs between feature work and infrastructure needs.
+
+- Further Reading
+  - http://artsy.github.io/blog/2015/12/22/future-fridays/
+  - https://lethain.com/migrations/
+
+### Done Means Done
+
+Not that when you tell someone it's done that everything is final. More that you can't just throw something over a
+fence, shipping code is one part of an equation but running QA and ensuring stability is just as important. Done
```

Comment:
The first part of the second sentence should probably be part of the first sentence instead of the second.